qemu/tests/migration_with_sgx.py

```diff
@@ -0,0 +1,52 @@
+from virttest import error_context
+from virttest.utils_misc import verify_dmesg
+
+from provider.sgx import SGXHostCapability
+from provider.sgx import SGXChecker
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Do migration test with sgx enabled
+
+    1. Check sgx and numa capability
+    2. Boot src and dst sgx guests with numa
+    3. Migration
+    4. Check dst VM sgx enabled in guest
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment
+    """
+
+    error_context.context("Start sgx test", test.log.info)
+    timeout = float(params.get_numeric("login_timeout", 240))
+
+    sgx_cap = SGXHostCapability(test, params)
+    sgx_cap.validate_sgx_cap()
+    host_total_epc_size = sgx_cap.host_epc_size
+    if params.get("monitor_expect_nodes"):
+        sgx_cap.validate_numa_node_count()
+
+    vm = env.get_vm(params["main_vm"])
+    vm.create()
+    vm.verify_alive()
+    vm.wait_for_login(timeout=timeout)
+
+    # do migration
+    error_context.context("Start to do sgx vm migration", test.log.info)
+    mig_timeout = int(params.get("mig_timeout", "3600"))
```

Comment:
we may also use params.get_numeric, just what we did in line 24

Reply:
agree with Zhenchao, params.get_numeric("mig_timeout", 3600)

Reply:
Updated thanks